root/static/scripts/common/components/Autocomplete2/formatters.js

```diff
@@ -0,0 +1,425 @@
+/*
+ * @flow
+ * Copyright (C) 2019 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import * as React from 'react';
+
+import CountryAbbr from '../../../../../components/CountryAbbr';
+import {commaOnlyListText} from '../../i18n/commaOnlyList';
+import {unwrapNl} from '../../i18n';
+import {addColonText} from '../../i18n/addColon';
+import {reduceArtistCredit} from '../../immutable-entities';
+import bracketed, {bracketedText} from '../../utility/bracketed';
+import formatDate from '../../utility/formatDate';
+import formatDatePeriod from '../../utility/formatDatePeriod';
+import formatTrackLength from '../../utility/formatTrackLength';
+
+import type {
+  AutocompleteAreaT,
+  AutocompleteEventT,
+  AutocompleteInstrumentT,
+  AutocompletePlaceT,
+  AutocompleteRecordingT,
+  AutocompleteReleaseT,
+  AutocompleteReleaseGroupT,
+  AutocompleteSeriesT,
+  AutocompleteWorkT,
+  Item,
+} from './types';
+
+const nonLatinRegExp = /[^\u0000-\u02ff\u1E00-\u1EFF\u2000-\u207F]/;
+
+function isNonLatin(str) {
+  return nonLatinRegExp.test(str);
+}
+
+function showExtraInfo(children, className = 'comment') {
+  return (
+    <span className={`autocomplete-${className}`}>
+      {children}
+    </span>
+  );
+}
+
+function showBracketedTextInfo(comment) {
+  return comment ? showExtraInfo(bracketedText(comment)) : null;
+}
+
+function showExtraInfoLine(children, className = 'comment') {
+  return (
+    <>
+      <br />
+      {showExtraInfo(children, className)}
+    </>
+  );
+}
+
+function formatGeneric(entity, extraInfo) {
+  const name = unwrapNl<string>(entity.name);
```

Comment:
If `formatGeneric` is called from `formatItem`, then `unwrapNl<string>(entity.name)` is computed twice.

Reply:
From what I can see, `formatItem` only calls `unwrapNl` inside the `if (!item.entityType) { ... }` block at the top, where it also returns, so doesn't reach `formatGeneric` below. So not sure I see that. (I guess it wouldn't matter either way, since `unwrapNl` is a no-op if the argument is already a string.)

Reply:
Yup, sorry about for misreading (twice) the code here.

Reply:
No need to be sorry, my code is not as clear as it could be (and a point of reviewing is figuring out what the heck it does). :)